test/modalWindow.spec.js

```diff
@@ -0,0 +1,39 @@
+import { ModalWindow } from '../src/app/layouts/ModalWindow';
+import '@testing-library/jest-dom';
+import { screen } from '@testing-library/dom';
+
+describe('Function that creates,show and close modal window', () => {
+  const parent = document.createElement('div');
+  parent.setAttribute('data-testid', 'test-parent');
+  const modalWindow = ModalWindow(parent);
+  const testContent = document.createElement('div');
+  testContent.setAttribute('data-testid', 'test-content');
+  testContent.textContent = 'test content';
+
+  it('Should display modal window with content', () => {
+    modalWindow.show(testContent);
+    document.body.innerHTML = parent.innerHTML;
+
+    expect(modalWindow).toBeVisible();
+    expect(screen.queryByTestId('test-content')).toHaveTextContent(
+      'test content',
+    );
+  });
+
+  it('Should hide modal window and delete content', () => {
+    const overlay = modalWindow.firstChild;
+    overlay.setAttribute('data-testid', 'test-overlay');
+    document.body.innerHTML = parent.innerHTML;
+
+    // here modalWindow contain content
```

Comment:
Jak widzę tutaj są jakieś problemy. Ten test jest mało czytelny - szczególnie to co się na górze dzieje.
Co powiesz na coś takiego :) ? 


```
import { ModalWindow } from '../src/app/layouts/ModalWindow';
import '@testing-library/jest-dom';
import { screen } from '@testing-library/dom';

describe('Modal Window', () => {

  it('when shown, then overlay and the content should be visible', () => {
    const modalContent = testContent();
    const modalWindow = testModalWindow((parent) => ModalWindow(parent));

    modalWindow.show(modalContent);

    expect(screen.queryByTestId('test-content')).toBeVisible();
    expect(screen.queryByTestId('test-modal-overlay')).toBeVisible();
    expect(screen.queryByTestId('test-modal-window')).toBeVisible();
  });

  it('when closed, then content should not visible', () => {
    const modalContent = testContent();
    const modalWindow = shownTestModalWindow((parent) => ModalWindow(parent), { content: modalContent });

    modalWindow.close()

    expect(screen.queryByTestId('test-content')).not.toBeInTheDocument();
    expect(screen.queryByTestId('test-modal-overlay')).not.toBeVisible();
    expect(screen.queryByTestId('test-modal-window')).not.toBeVisible();
  });

});

function testModalWindow (modalFn) {
  document.body.innerHTML = `<div id="component-container"></div>`;
  const componentContainer = document.getElementById('component-container');
  const modalWindow = modalFn(componentContainer);
  modalWindow.setAttribute('data-testid', 'test-modal-window');
  const modalOverlay = modalWindow.querySelector('#overlay');
  modalOverlay.setAttribute('data-testid', 'test-modal-overlay')
  return modalWindow;
}

function shownTestModalWindow (modalFn, { content }) {
  const testModal = testModalWindow(modalFn);
  testModal.show(content);
  return testModal;
}

function testContent () {
  const testContent = document.createElement('div');
  testContent.setAttribute('data-testid', 'test-content');
  testContent.textContent = 'test content text';
  return testContent;
}

```

Reply:
Przydałby się jeszcze taki test:
```
  it('when created, then should not be visible', () => {
    const modalWindow = testModalWindow((parent) => ModalWindow(parent));

    expect(screen.queryByTestId('test-modal-overlay')).not.toBeVisible();
    expect(screen.queryByTestId('test-modal-window')).not.toBeVisible();
  });
```
Ale znalazłem z tym pewien problem. Testing-library nie bierze pod uwage plików css. Czyli jesli chcemy, zeby to zadziałało to przy tworzeniu modala w JSie trzeba ustawic
`modalWindow.style.display = 'none'`

Reply:
W sumie jeszcze brakuje też testu co powinno się stać jak ktoś wywoła close() za nie wyświetlonym modalu :)

Reply:
![image](https://user-images.githubusercontent.com/31566345/104177873-c450ca80-5409-11eb-83f2-a90eb005f17e.png)
a to nie pokazuje, że jednak metoda toBeVisible sprawdza css?

Reply:
No wlasnie sprawdza **css property**. Czyli jak ustawisz property to będzie OK. Tutaj ono się nie ustawia, bo wszystko, co tu masz jest w JSie, czyli masze pliki scss nie są brane pod uwagę. 

W sumie fajnie by było, żeby były. Ale nie wiem jak i czy jest możliwe. Może ustawienia JESTa jakos?

Reply:
![image](https://user-images.githubusercontent.com/31566345/104178591-e565eb00-540a-11eb-8e24-4b89d518c8f1.png)
chodzi o to, że tutaj w divie by musiało być style="display:none, opacity:0", tak ?

Reply:
a dobra teraz zrozumiałem co napisałeś ...., że nie sprawdza plików css, a nie stylów css ...